packages/itwin/tree-widget/README.md

```diff
@@ -393,3 +393,129 @@ function MyWidget() {
   );
 }
 ```
+
+For custom tree components `TelemetryContextProvider` should be used:
+
+```tsx
+import { TelemetryContextProvider } from "@itwin/tree-widget-react";
+
+function MyWidget() {
+  return <TelemetryContextProvider
+    componentIdentifier="MyTree"
+    onPerformanceMeasured={(feature, elapsedTime) => {
+      console.log(`TreeWidget [${feature}] took ${elapsedTime} ms`)
+    }}
+    onFeatureUsed={(feature) => {
+      console.log(`TreeWidget [${feature}] used`)
+    }}
+  >
+    <MyTree />
+  </TelemetryContextProvider>;
+}
+
+function MyTree() {
+  // see "Custom trees" section for example implementation
+}
+```
+
+## Custom trees
```

Comment:
I'd move this section as `###` under `## Components`, between sections `iModel content tree` and `Hierarchy level size limiting`.